browser/resources/brave_new_tab_page_refresh/components/search/search_results.tsx

```diff
@@ -17,10 +19,11 @@ import { SafeImage } from '../common/safe_image'
 
 import { style } from './search_results.style'
 
-function MatchImage(props: { match: SearchResultMatch }) {
+function MatchImage(props: { match: AutocompleteMatch }) {
   const { getString } = useLocale()
   const { imageUrl, iconUrl } = props.match
-  if (props.match.description === getString('searchAskLeoDescription')) {
+  const description = mojoString16ToString(props.match.description)
```

Comment:
I wonder if it makes sense to memoize this call on `match` - I have no idea how heavy `mojoString16ToString` is but seems like some low hanging fruit:

`const description = useMemo(() => mojoString16ToString(props.match.description), [props.match])`

Reply:
I'm actually not sure. I think React actually rerenders all child components by default on rerender (even if the props are unchanged) unless they're wrapped in a `React.memo`

https://react.dev/reference/react/memo

edit: just put together a test - it will rerender every time the parent changes:
https://playcode.io/2370544

Reply:
Ah - thanks for checking and the docs reference. Updated to use `useMemo`.